ksql-engine/src/test/java/io/confluent/ksql/codegen/CodeGenRunnerTest.java

```diff
@@ -63,9 +112,288 @@ private Analysis analyzeQuery(String queryStr) {
+    }
+
+    private boolean evalBooleanExprNeq(int cola, int colb, Object values[]) throws Exception {
+        return evalBooleanExpr("SELECT col%d != col%d FROM CODEGEN_TEST;", cola, colb, values);
+    }
+
+    private boolean evalBooleanExprIsDistinctFrom(int cola, int colb, Object values[]) throws Exception {
+        return evalBooleanExpr("SELECT col%d IS DISTINCT FROM col%d FROM CODEGEN_TEST;", cola, colb, values);
+    }
+
+    private boolean evalBooleanExprLessThan(int cola, int colb, Object values[]) throws Exception {
+        return evalBooleanExpr("SELECT col%d < col%d FROM CODEGEN_TEST;", cola, colb, values);
+    }
+
+
+    private boolean evalBooleanExprLessThanEq(int cola, int colb, Object values[]) throws Exception {
+        return evalBooleanExpr("SELECT col%d <= col%d FROM CODEGEN_TEST;", cola, colb, values);
+    }
+
+    private boolean evalBooleanExprGreaterThan(int cola, int colb, Object values[]) throws Exception {
+        return evalBooleanExpr("SELECT col%d > col%d FROM CODEGEN_TEST;", cola, colb, values);
+    }
+
+    private boolean evalBooleanExprGreaterThanEq(int cola, int colb, Object values[]) throws Exception {
+        return evalBooleanExpr("SELECT col%d >= col%d FROM CODEGEN_TEST;", cola, colb, values);
+    }
+
+    @Test
+    public void testNullEquals() throws Exception {
+        Assert.assertFalse(evalBooleanExprEq(5, 0, new Object[]{null, 12344L}));
+        Assert.assertFalse(evalBooleanExprEq(5, 0, new Object[]{null, null}));
+    }
+
+    @Test
+    public void testIsDistinctFrom() throws Exception {
+        Assert.assertFalse(evalBooleanExprIsDistinctFrom(5, 0, new Object[]{12344, 12344L}));
+        Assert.assertTrue(evalBooleanExprIsDistinctFrom(5, 0, new Object[]{12345, 12344L}));
+        Assert.assertTrue(evalBooleanExprIsDistinctFrom(5, 0, new Object[]{null, 12344L}));
+        Assert.assertFalse(evalBooleanExprIsDistinctFrom(5, 0, new Object[]{null, null}));
+    }
+
+    @Test
+    public void testIsNull() throws Exception {
+        String simpleQuery = "SELECT col0 IS NULL FROM CODEGEN_TEST;";
+        Analysis analysis = analyzeQuery(simpleQuery);
+
+        ExpressionMetadata expressionEvaluatorMetadata0 = codeGenRunner.buildCodeGenFromParseTree
+            (analysis.getSelectExpressions().get(0));
+        Assert.assertTrue(expressionEvaluatorMetadata0.getIndexes().length == 1);
+        int idx0 = expressionEvaluatorMetadata0.getIndexes()[0];
+        Assert.assertEquals(idx0, 0);
+        Assert.assertEquals(expressionEvaluatorMetadata0.getUdfs().length, 1);
+
+        Object result0 = expressionEvaluatorMetadata0.getExpressionEvaluator().evaluate(new Object[]{null});
+        assertThat(result0, instanceOf(Boolean.class));
+        Assert.assertTrue((Boolean)result0);
+
+        result0 = expressionEvaluatorMetadata0.getExpressionEvaluator().evaluate(new Object[]{12345L});
+        assertThat(result0, instanceOf(Boolean.class));
```

Comment:
Maybe just standardize on the richer hamcrest asserts, rather than mixing both? You can always do `assertThat(blah, is(true))`